bip-psbt.mediawiki

```diff
@@ -0,0 +1,275 @@
+  Status: Draft
+  Type: Standard
+  Created: 2017-07-12
+  License: BSD-2-Clause
+</pre>
+
+==Introduction==
+
+===Abstract===
+
+This document proposes a binary transaction format which contains the information
+necessary for a signer to produce signatures for the transaction and holds the
+signatures for an input while the input does not have a complete set of signatures.
+The signer can be offline as all necessary information will be provided in the
+transaction.
+
+===Copyright===
+
+This BIP is licensed under the 2-clause BSD license.
+
+===Motivation===
+
+Creating unsigned or partially signed transactions to be passed around to multiple
+signers is currently implementation dependent, making it hard for people who use
+different wallet software from being able to easily do so. One of the goals of this
+document is to create a standard format that can be used between clients to allow
+people to pass around the same transaction to sign and combine their signatures.
+
+Signing transactions also requires users to have access to the blockchain in order
+to find the UTXOs being spent so that the transaction can be signed. This transaction
+format will allow offline signers such as air-gapped wallets and hardware wallets
+to be able to sign transactions without access to the blockchain.
+
+==Specification==
+
+The transaction format uses a key-value map. Each key-value pair is as follows:
+
+{| class="wikitable" style="width: auto; text-align: center; font-size: smaller; table-layout: fixed;"
+!Name
+!Type
+!Description
+|-
+| Key Length
+| Compact Size Unsigned Integer
+| Specify how long the key is
+|-
+| Key
+| byte[]
+| The key itself
+|-
+| Value Length
+| Compact Size Unsigned Integer
+| Specify how long the value is
+|-
+| Value
+| byte[]
+| The Value itself
+|}
+
+The first byte of each key specifies the type of the key-value pair. Smoe types are
+for global fields and other fields are for each input. The currently
+defined global types are as follows:
+
+{| class="wikitable" style="width: auto; text-align: center; font-size: smaller;
+table-layout: fixed;"
+!Number
+!Name
+!Key Data
+!Value Data
+|-
+| <tt>0x00</tt>
+| Transaction
+| None. The key must only contain the 1 byte type.
+| The transaction in network serialization. The scriptSigs and
+witnesses for each input must be empty unless the input is finalized. The transaction
+must be in the witness serialization format as defined in BIP 144.
+|-
+| <tt>0x01</tt>
+| Redeem Script
+| The hash160 of the redeem script
+| A redeem script that will be needed to sign a Pay-To-Script-Hash input.
+|-
+| <tt>0x02</tt>
+| Witness Script
+| The sha256 hash of the witness script
+| A witness script that will be needed to sign a Pay-To-Witness-Script-Hash input.
+|}
+
+The currently defined per-input types are defined as follows:
+
+{| class="wikitable" style="width: auto; text-align: center; font-size: smaller;
+table-layout: fixed;"
+!Number
+!Name
+!Key Data
+!Value Data
+|-
+| <tt>0x03</tt>
```

Comment:
Perhaps it makes sense to reuse the same type bytes as used in global, to avoid having implementations that accept globals in the input-specific blocks or the other way around?